### PR TITLE
fix: AUT-26 show an item as informational if it does not have response declarations

### PR DESCRIPTION
**Pull request by pnal**
https://oat-sa.atlassian.net/browse/AUT-26
AC: the item should be considered as informational automatically if it has A-block in it.

### CODE REVIEW

File: models/test/mapper/TestPreviewMapper.php [hunk 1]
```diff
@@ -230,11 +231,16 @@ private function getService(): Service
...
+     * @param $itemRef
      * @return bool
      */
-    private function isItemInformational($categories): bool
+    private function isItemInformational($categories, $itemRef): bool
     {
-        return in_array('x-tao-itemusage-informational', $categories, true);
+        $additionalCheck = false;
+
+        if ($itemRef instanceof ExtendedAssessmentItemRef && !count($itemRef->getResponseDeclarations())){
```
Comment (pnal) [inline models/test/mapper/TestPreviewMapper.php:241]: the same as in the test runner: https://github.com/oat-sa/extension-tao-testqti/blob/master/helpers/class.TestRunnerUtils.php#L734

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/test/mapper/TestPreviewMapper.php [hunk 2]
```diff
@@ -230,11 +231,16 @@ private function getService(): Service
...
      * @param $categories
-     *
+     * @param $itemRef
      * @return bool
      */
-    private function isItemInformational($categories): bool
+    private function isItemInformational($categories, $itemRef): bool
     {
-        return in_array('x-tao-itemusage-informational', $categories, true);
+        $additionalCheck = false;
```
Comment (mike-ionut-mihai-sandu-tao) [inline models/test/mapper/TestPreviewMapper.php:239]: do we need to modify tests ?

Comment (mike-ionut-mihai-sandu-tao) [reply models/test/mapper/TestPreviewMapper.php:239]: \oat\taoQtiTestPreviewer\test\unit\models\test\factory\TestPreviewMapperTest maybe?

File: test/unit/models/test/mapper/TestPreviewMapperTest.php [hunk 3]
```diff
@@ -81,57 +82,105 @@ public function testMapEmptyTest(): void
...
+                            'label' => $sectionTitle,
+                            'isCatAdaptive' => false,
                             'position' => 0,
-                            'isLinear' => true,
-                            'sections' => [
-                                'sectionId' => [
-                                    'id' => 'sectionId',
-                                    'label' => $sectionTitle,
-                                    'isCatAdaptive' => false,
+                            'items' => [
```
Comment (pnal) [inline test/unit/models/test/mapper/TestPreviewMapperTest.php:172]: Diff looking weird but I just moved out the array assignment for re-using

File: models/test/mapper/TestPreviewMapper.php (hunk 2 above)
Comment (bartlomiejmarszal) [reply models/test/mapper/TestPreviewMapper.php:239]: You are introducing parameter that is not sharing common interface and you are not declaring type in parameter. This seems sketchy.

**Review by bartlomiejmarszal — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: test/unit/models/test/mapper/TestPreviewMapperTest.php (hunk 3 above)
Comment (bartlomiejmarszal) [reply test/unit/models/test/mapper/TestPreviewMapperTest.php:172]: Yes. I wish you could extract code refactoring from your actual changes. This would be much easier to read.

File: models/test/mapper/TestPreviewMapper.php [hunk 4]
```diff
@@ -230,11 +231,16 @@ private function getService(): Service
...
+     * @param AssessmentItemRef|ExtendedAssessmentItemRef $itemRef
      * @return bool
      */
-    private function isItemInformational($categories): bool
+    private function isItemInformational($categories, $itemRef): bool
     {
-        return in_array('x-tao-itemusage-informational', $categories, true);
+        $additionalCheck = false;
+
+        if (method_exists($itemRef, 'getResponseDeclarations') && !count($itemRef->getResponseDeclarations())) {
```
Comment (shpran) [inline models/test/mapper/TestPreviewMapper.php:241]: Can we use `empty` here?
```
if (method_exists($itemRef, 'getResponseDeclarations') && !count($itemRef->getResponseDeclarations())) {
```
========>
```
if (method_exists($itemRef, 'getResponseDeclarations') && empty($itemRef->getResponseDeclarations())) {
```

**Review by shpran — APPROVED** [review]
Good job! 👍🏻 

- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/test/mapper/TestPreviewMapper.php (hunk 2 above)
Comment (pnal) [reply models/test/mapper/TestPreviewMapper.php:239]: @bartlomiejmarszal 
Well, I see all the code is relying on` AssesmentItemRef` but actually https://github.com/oat-sa/qti-sdk/blob/master/src/qtism/runtime/tests/RouteItem.php#L154 in spite of phpDoc returns `ExtendedAssessmentItemRef` class object and it gives us an ability to check does the item have response declarations. I can not change/set type declarations because as I understand it depends on QTI content version and can brake compatibility, but in practice it solves this particular issue (testPreview do not automatically mark items informational although testRunner do) so I think we can go with it

File: test/unit/models/test/mapper/TestPreviewMapperTest.php (hunk 3 above)
Comment (pnal) [reply test/unit/models/test/mapper/TestPreviewMapperTest.php:172]: Sorry for that, unit tests carried me away ;)

File: models/test/mapper/TestPreviewMapper.php (hunk 2 above)
Comment (julien-sebire) [reply models/test/mapper/TestPreviewMapper.php:239]: ExtendedAssessmentItemRef actually extends AssessmentItemRef.
It seems that for practical reasons, ExtendedAssessmentItemRef's are stored in test session instead of AssessmentItemRef's so I think you can safely rely on it. I'll include this phpdoc change in a future release.

File: models/test/mapper/TestPreviewMapper.php (hunk 4 above)
Comment (julien-sebire) [reply models/test/mapper/TestPreviewMapper.php:241]: Just by curiosity, why use empty instead of count? empty is less precise than count and doesn't say what we're testing.
All the more, if you want to test that `$itemRef->getResponseDeclarations()` is an empty array, why not doing it explicitly with
```
$itemRef->getResponseDeclarations() === []
```
?

**Comment by AntonNeviarouski** [comment]
QA verified on http://test-nik.playground.kitchen.it.taocloud.org:48161/
